### ISSUE TITLE
False positive highlighting in sub-patterns

### ISSUE THREAD

**Issue by ilyavenner** [intellij-rust/intellij-rust#7530]
## Environment

* **IntelliJ Rust plugin version:** 0.4.150.3968-211
* **Rust toolchain version:** 1.52.1 (9bc8c42bb 2021-05-09) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2021.1.3 (CL-211.7628.27)
* **Operating system:** Linux 5.10.42-1-manjaro
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

A directly imported variant of an enumeration is highlighted as an unused variable in `match` sub-patterns (see line `9`).

![image](https://user-images.githubusercontent.com/13315957/125662395-8a2a4e6e-075e-4721-b73a-24ddcf91e5cc.png)

## Steps to reproduce

To reproduce try this code:

```rust
fn main() {
    use Foo::*;

    let foo = Foo::Bar;

    match foo {
        f @ Bar => println!("bar"),
        Baz => println!("baz"),
    }
}

enum Foo {
    Bar,
    Baz,
}
```
